packages/components/src/PieChartButton/PieChartButton.component.js

```diff
@@ -0,0 +1,387 @@
+		outerRadius: size.outerRadius,
+		padAngle: size.padAngle,
+		startAngle: allPercentages * 2 / 100 * Math.PI,
+		endAngle: Math.PI * 2,
+	});
+	return (
+		<path
+			d={arcGenerated}
+			transform={`translate(${size.svgSize / 2},${size.svgSize / 2})`}
+			stroke={hover ? SCOOTER : emptyColor}
+			fill={hover ? SCOOTER : emptyColor}
+		/>
+	);
+}
+
+/**
+ * This function generate a part of circle for a value
+ * @param {object} value the value to generate
+ * @param {number} index the current index
+ * @param {array} values the values to get when we have to start
+ * @param {object} size the current graph size
+ * @param {function} arcGen the arc generator
+ * @param {boolean} hover known if the component is hovered
+ */
+export function getCircle(value, index, values, size, arcGen, hover) {
+	let percentagesDone = 0;
+	for (let i = 0; i < index; i += 1) {
+		percentagesDone += values[i].percentageShown;
+	}
+
+	const arcGenerated = arcGen({
+		innerRadius: size.innerRadius,
+		outerRadius: size.outerRadius,
+		padAngle: size.padAngle,
+		startAngle: percentagesDone * 2 / 100 * Math.PI,
+		endAngle: (percentagesDone + value.percentageShown) * 2 / 100 * Math.PI,
+	});
+
+	return (
+		<path
+			key={index}
+			d={arcGenerated}
+			transform={`translate(${size.svgSize / 2},${size.svgSize / 2})`}
+			stroke={hover ? SCOOTER : value.color}
+			fill={hover ? SCOOTER : value.color}
+		/>
+	);
+}
+
+/**
+ * This function is just a sort
+ * @param {object} a an element with a percentageShown to compare
+ * @param {object} b an other element with a percentageShown to compare
+ */
+function sortElements(a, b) {
+	if (a.percentageShown > b.percentageShown) return -1;
+	if (a.percentageShown < b.percentageShown) return 1;
+	return 0;
+}
+
+/**
+ * This function set percentage show with the minimum percentage shown
+ * @param {array} values the set of values
+ * @param {number} minimumPercentage the minimum value we have to show
+ */
+export function setMinimum(model, minimumPercentage) {
+	let amountToSubtract = 0;
+
+	const valuesMins = model.map(value => {
+		if (value.percentage < minimumPercentage && value.percentage > 0) {
+			amountToSubtract += minimumPercentage - value.percentage;
+			return { ...value, percentageShown: minimumPercentage };
+		}
+		return { ...value, percentageShown: value.percentage };
+	});
+
+	/**
+	 * This function decrease the percentage shown & the amount to subtract by 1
+	 * @param {object} element the current element
+	 */
+	function decreaseElement(element) {
+		if (amountToSubtract > 0) {
+			// eslint-disable-next-line no-param-reassign
+			element.percentageShown -= 1;
+			amountToSubtract -= 1;
+		}
+	}
+
+	while (amountToSubtract > 0) {
+		const elementsToDecrease = valuesMins
+			.filter(value => value.percentageShown > minimumPercentage)
+			.sort(sortElements);
+		if (elementsToDecrease.length > 0) {
+			elementsToDecrease.forEach(decreaseElement);
+		} else {
+			amountToSubtract = 0;
+		}
+	}
```

Comment:
Should be replaced by a `reduce` on `elementToDecrease`. Without the `while` loop.